src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1648,19 +1648,31 @@ RouterInsertTaskList(Query *query, bool parametersInQueryResolved,
 
 	Assert(query->commandType == CMD_INSERT);
 
+	/*
+	 * Multi-row inserts may be large and BuildRoutesForInsert allocates some
```

Comment:
Isn't this actually increasing the peak memory usage at this stage? Now it is:

1) allocate things on `BuildRoutesForInsert`
2) copy `modifyRoute->rowValuesLists`
3) deallocate   `BuildRoutesForInsert`

On the master, we didn't have (2). 

My question is that are we sure that `BuildRoutesForInsert` allocates significant memory other than `modifyRouteList`? In other words, is it worth having this context given the increase until we clear the context ?